WalletWasabi/Crypto/ZeroKnowledge/NonInteractive/Prover.cs

```diff
@@ -0,0 +1,53 @@
+using NBitcoin.Secp256k1;
+using System.Collections.Generic;
+using System;
+using System.Linq;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Crypto.ZeroKnowledge.LinearRelation;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge.NonInteractive
+{
+	public static class Prover
```

Comment:
Since this is a single method static class, does it make sense to fold into `ProofSystem` now? Same goes for `Verifier`

Reply:
I don't think so.